app/src/main/java/com/alphawallet/app/ui/FunctionActivity.java

```diff
@@ -174,6 +174,12 @@ private void getAttrs()
 
         // Fetch attributes local to this action and add them to the injected token properties
         Map<String, TSAction> functions = viewModel.getAssetDefinitionService().getTokenFunctionMap(token.tokenInfo.chainId, token.getAddress());
+        if (functions == null)
```

Comment:
Doesn't seem related to the PR/commit?

Reply:
Doe this effectively do a restart/refresh of the activity?

Reply:
> Doe this effectively do a restart/refresh of the activity?

Yes, it's an OS directive to do just that. It preserves the starting intents too - so the app will probably just appear to 'flicker' then be in the required state. This ```recreate()``` was found during intensive usage while I was flicking between the app and other apps. The other ```recreate()``` simplifies some other null checking code. This one probably should go into a separate PR.